--- a/conf.c
+++ b/conf.c
@@ -39,7 +39,7 @@
 	conf->database = 0;
 	conf->pool_size_per_thread = 2;
 
-	j = json_load_file(filename, 0, &error);
+	j = json_load_file(filename, &error);
 	if(!j) {
 		fprintf(stderr, "Error: %s (line %d)\n", error.text, error.line);
 		return conf;
--- webdis-0.1.0.orig/formats/json.c
+++ webdis-0.1.0/formats/json.c
@@ -227,7 +227,7 @@ json_ws_extract(struct http_client *c, c
 
 	jsonz = calloc(sz + 1, 1);
 	memcpy(jsonz, p, sz);
-	j = json_loads(jsonz, sz, &jerror);
+	j = json_loads(jsonz, &jerror);
 	free(jsonz);
 
 	if(!j) {
